website_product_filters/controllers/main.py

```diff
@@ -135,6 +144,20 @@ def _normalize_category(self, category):
             cond = category.id
         return cond
 
+    def _get_brand_categories(self, brand_ids):
+        cr, uid, context, pool = (request.cr,
+                                  request.uid,
+                                  request.context,
+                                  request.registry)
+        categ_ids = []
+        product_obj = pool['product.template']
+        product_ids = product_obj.search(
+            cr, uid, [('product_brand_id', 'in', brand_ids)])
+        for product in product_obj.browse(cr, uid, product_ids, context):
+            if product.public_categ_ids not in categ_ids:
+                categ_ids.append(product.public_categ_ids)
```

Comment:
when you use a object from a request `request.registry` it brings you old API objects, in your code you are asking to call the chained method `.mapped()` from a list object. besides I'm not returning a list of  ids, im returning a list of objects.